### DISCUSSION TITLE
How do I make the sidebar appear?

### DISCUSSION THREAD

**Discussion by Kajakster** [datenstrom/community#645] [Ask a question] [answered]
Hello,
I just started with yellow with a fresh install. I like the idea of a simple website. 

I do not have any coding or Web development background. 
This is a fresh install, I selected "blog" during Installation. 

However, I can not get the sidebar to show. I uploaded a sidebar.md to the shared folder (content as in the example in the git hub documentation) , and also to the blog folder, but they don't show up on the website. Samegoes for header and footer respectively . 

Is there anything I have forgotten? Do I have to edit the page.md for activating the Sidebar?

Code of sidebar.md:
```
---
Title: Sidebar
Status: shared
---
Neu

[blogchanges /blog/]

Tags

[blogtags /blog/]
```

**Comment by nzyeehaa** [comment, marked as answer]
learning by doing is fantastic, but can be a steep curve

I will throw you a few crumbs to help the learning....

it looks like you are doing everything correctly again - well done...
on a brief investigation - it looks like the default theme doesnt include sidebar... ( fyi @markseu )
{...or navigation-tree as you discovered already}

try [berlin theme](https://github.com/datenstrom/yellow-extensions/tree/master/source/berlin) which includes sidebar layout (see berlin-default.html)

and the basic code that brings it in
`<?php echo $this->yellow->page->getPage("sidebar")->getContent() ?>`

p.s: you should of course include a check to make sure a sidebar page exists - or your site may break when there was none

**Reply by Kajakster** [reply]
Thanks again for the Input.

I somehow get the Idea that yellow is targeted mainly to developers or people with coding experience. 

Maybe it is clear for coders that the theme has to include the sidebar. I just had the Idea that if I did it as described it would just work. 

So I will Install the other theme then. 

I just have to find out how to update the documentation on these topics.

**Reply by nzyeehaa** [reply]
every new software has a learning curve, is in different stages of development, and is managed in different ways.

you followed the instructions well, but just got caught up in some changes- hang in there.

Open source software is always in the need of assistance and developers, and Yellow has a very small team - you can [read about contributing here](https://datenstrom.se/yellow/help/contributing-guidelines)

The sidebar used to be included in the templates, but has either been forgotten, or missed out for some reason and the documentation not updated to cater for the omission - I am not sure which. You struck it lucky as I was looking for some answers and saw your unanswered question so stopped to help out.

To be clear, I am not part of the yellow team, with the exception that I love it for many reasons, and have been using it for several years. It is still in its infancy, but is doing really well,
While its aim is to be easy to use, I perceive there are five levels to any web software, {developer, webmaster, designer, documentation, user} and people tend to span 1 or 2 areas fairly competently, sometimes more.

I too can get frustrated and can waste a lot of time troubleshooting when things don't go as advertised - but it is free software after all. I have made, used and contributed to many platforms over the last 30+ years and aim to stick with yellow and I regularly convert websites from more complex CMS's. I have learnt to manage it to avoid hiccups, and customize it to fit my needs/standards.
You can even do your own documentation if the sites is not fitting the bill, but when more people contribute, the software can move faster.

Hope that helps in some way.

**Reply by Kajakster** [reply]
By changing the theme to "Berlin" the Sidebars appeared. 

But not on the Blog Page. 
But I managed to copy/paste the Code for the sidebar from the other pages to the blog.html and the blogpages.html
Now it works on the blogpage too.

**Reply by nzyeehaa** [reply]
good job, if any comments have helped - mark it as answered

**Reply by cellio** [reply]
Hello.  I'm also learning by doing -- new to SSGs and Yellow, and never used PHP before.  I tried adding the PHP line suggested here to both blog.html and blogpages.html, after installing the Berlin theme, but I'm not seeing a sidebar.  Where in those files am I supposed to put the line?  Do I need to do something else to turn the sidebar on?  I'd like to be able to figure this out for myself but I'm a little lost and don't quite know what I should be looking for yet -- thanks!

**Reply by nzyeehaa** [reply]
@cellio welcome to yellow....
the berlin theme already includes the code to show the sidebar, [see here...](https://github.com/datenstrom/yellow-extensions/blob/master/source/berlin/berlin-default.html)

So if it is still not working for you, you have either not added a sidebar or have added it incorrectly.

Kajakster added a sidebar.md to the shared folder.
Or you can add sidebar.md in a content folder

**Reply by cellio** [reply]
Thank you.  I see now.  A few things I figured out from this thread:

- Installing the Berlin theme doesn't change the layout; you have to copy berlin-default.html to default.html.
- When you edit blog.html and blogpages.html to add the sidebar, you need both of the first two <div>s, not just the one named "sidebar".
- Trying to edit the php line in the first comment into blog[pages].html was a red herring.

Slowly I'm learning.  Thanks for the help.

**Reply by tikita** [reply]
First I changed Yellow to the 'Berlin' theme.

Then I put a new sidebar.md in the folder content/shared.

Then I renamed the file berlin-default.html in the folder system/layouts to default.html (and removed the previous default.html before).

The new sidebar now appeared on the right (!) side of the called page. I would prefer the sidebar to be on the left, but I don't know how to do that yet.

**Comment by markseuffert** [comment]
Check what's written in the [documentation of themes](https://datenstrom.se/yellow/extensions/#themes). 

If a theme doesn't mention a sidebar, then it does **not** support it out-of-the-box. Datenstrom Yellow targets small websites, with less features and fewer distractions. As we're getting closer to updating themes, we might add theme(s) with two column-layout. It's a feature we are looking to provide. 

See also datenstrom/community#531.

**Reply by tikita** [reply]
_> As we're getting closer to updating themes, we might add theme(s)_
_> with two column-layout. It's a feature we are looking to provide._

I would be very happy if there was a theme that had the navigation not at the top but on the left side. And it would not be important that there is a two column-layout.

Many regards, Heiner
